### PR TITLE
fix: CMS content type check

### LINKED ISSUE #33: Fail fast on cms verifySignedAttributes
In cms package, when verifying the signed attributes, we can add a check on the content type as RFC 3161 requiring the value to be ` id-ct-TSTInfo`.

### PR DESCRIPTION
**Pull request by JeyJeyGao**
**What this PR does / why we need it**:
Previously, the CMS package was generically designed for RFC 5652. However, since the CMS package will not be exported and we only need to care about RFC 3161, we can restrict the CMS content type to TSTInfo (id-ct-TSTInfo) to fail fast.

**Which issue(s) this PR resolves** *(optional, in `resolves #<issue number>(, resolves #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Resolves #33

**Please check the following list**:
- [x]  Does the affected code have corresponding tests, e.g. unit test?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

File: internal/cms/signed.go [hunk 1]
```diff
@@ -248,14 +248,17 @@ func (d *ParsedSignedData) verifySignedAttributes(signerInfo *SignerInfo, chains
 
 	// verify attributes if present
 	if len(signerInfo.SignedAttributes) == 0 {
-		if d.ContentType.Equal(oid.Data) {
-			return nil, nil
-		}
-		// signed attributes MUST be present if the content type of the
-		// EncapsulatedContentInfo value being signed is not id-data.
+		// According to RFC 5652, if the Content Type is id-data, signed
+		// attributes can be empty. However, the CMS package is designed for
```
Comment (shizhMSFT) [inline internal/cms/signed.go]:
```suggestion
		// attributes can be empty. However, this cms package is designed for
```

File: internal/cms/signed.go [hunk 2]
```diff
@@ -248,14 +248,17 @@ func (d *ParsedSignedData) verifySignedAttributes(signerInfo *SignerInfo, chains
...
+		// timestamp (RFC 3161) and the content type must be id-ct-TSTInfo,
+		// so we require signed attributes to be present.
 		return nil, VerificationError{Message: "missing signed attributes"}
 	}
 
+	// The content type must be id-ct-TSTInfo for timestamp (RFC 3161).
```
Comment (shizhMSFT) [inline internal/cms/signed.go]: Can you also have the comment that this block of check is for the optimization of the tsp client?

File: internal/cms/signed.go (hunk 1 above)
Comment (JeyJeyGao) [reply internal/cms/signed.go]: Updated.

File: internal/cms/signed.go (hunk 2 above)
Comment (JeyJeyGao) [reply internal/cms/signed.go]: Updated.

File: internal/cms/signed.go [hunk 3]
```diff
@@ -248,14 +248,19 @@ func (d *ParsedSignedData) verifySignedAttributes(signerInfo *SignerInfo, chains
...
+		// timestamp (RFC 3161) and the content type must be id-ct-TSTInfo,
+		// so we require signed attributes to be present.
 		return nil, VerificationError{Message: "missing signed attributes"}
 	}
 
+	// this CMS package is designed for timestamping (RFC 3161), so checking the
+	// content type to be id-ct-TSTInfo is an optimization for tspclient to
+	// fail fast.
+	if !oid.TSTInfo.Equal(d.ContentType) {
+		return nil, fmt.Errorf("unexpected content type: %v", d.ContentType)
```
Comment (ghost) [inline internal/cms/signed.go]: Please also display what the expected content type is in the error msg.

Comment (JeyJeyGao) [reply internal/cms/signed.go]: Updated.

**Review by ghost — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**